Rakefile

```diff
@@ -229,3 +229,26 @@ task :verify_private_key_present do
 end
 
 task build: :verify_private_key_present
+
+desc "Updates the rspec.github.io docs"
+task :update_docs, [:version, :branch, :website_path] do |t, args|
+  abort "You must have ag installed to generate docs" if `which ag` == ""
+  args.with_defaults(:website_path => "../rspec.github.io")
+  sh "git checkout #{args[:branch]} && git pull --rebase"
+  cmd = "bundle install && \
+         RUBYOPT='-I#{args[:website_path]}/lib' bundle exec yard \
+                          --yardopts .yardopts \
+                          --plugin rspec-docs-template \
```

Comment:
Is there any difference if we remove this option? I remember having some troubles installing this plugin.

Reply:
Yes. For example If I remove it. To install it. You have to go to https://github.com/rspec/rspec.github.io, source branche and install the gem manually. I will add some documentation about it.

```diff
--- a/source/documentation/4.0/rspec-rails/top-level-namespace.html
+++ b/source/documentation/4.0/rspec-rails/top-level-namespace.html
@@ -1,7 +1,8 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <meta charset="utf-8">
+<meta name="viewport" content="width=device-width, initial-scale=1.0">
 <title>
   Top Level Namespace

@@ -9,11 +10,11 @@

 </title>

-  <link rel="stylesheet" href="/documentation/4.0/rspec-rails/css/style.css" type="text/css" charset="utf-8" />
+  <link rel="stylesheet" href="/documentation/4.0/rspec-rails/css/style.css" type="text/css" />

-  <link rel="stylesheet" href="/documentation/4.0/rspec-rails/css/common.css" type="text/css" charset="utf-8" />
+  <link rel="stylesheet" href="/documentation/4.0/rspec-rails/css/common.css" type="text/css" />

-  <link rel="stylesheet" href="/documentation/4.0/rspec-rails/css/rspec.css" type="text/css" charset="utf-8" />
+  <link rel="stylesheet" href="/documentation/4.0/rspec-rails/css/rspec.css" type="text/css" />

 <script type="text/javascript">
   pathId = "";
@@ -101,22 +102,11 @@
 </div>

       <div id="footer">
-  Generated on Tue Mar 24 23:08:28 2020 by
+  Generated on Mon Mar 30 14:29:40 2020 by
   <a href="http://yardoc.org" title="Yay! A Ruby Documentation Tool" target="_parent">yard</a>
   0.9.24 (ruby-2.7.0).
 </div>

-<script>
-  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
-  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
-  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
-  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
-
-  ga('create', 'UA-59095215-1', 'auto');
-  ga('send', 'pageview');
-
-</script>
-
```

Reply:
Honestly if those are the only changes we can ditch it, it was supposed to do what the `ag` command does now but I never made it workable.

Reply:
I'm debating wether we really want to do this at all, I mean it could still be managed from the rspec-dev tools with a bit of refactoring...

Reply:
As you. We can have command for `rspec-rails` documentation and another one for other project. I am not opposed to this proposal.

Reply:
I think I'd rather we have a task from the dev repo to do this, down the line maybe we can extract a common dependency

Reply:
## Yard templates
I dig into a little bit, and the template does a bit more.

It choose different templates for the header and the footer.

### Google Analytics
For example with this Google Analytics snippet https://github.com/rspec/rspec.github.io/pull/44/commits/7b9add77a9f82adf5e5f1e1eb50f06ccda98fb2a

### Header properties

Also the header is a little bit different.
```diff
-    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <meta charset="utf-8">
+<meta name="viewport" content="width=device-width, initial-scale=1.0">
```

We will use [the default one from yard](https://github.com/lsegal/yard/blob/258821c9af712767d43bdf02eb48892955846785/templates/default/layout/html/headers.erb) if we remove the plugin. How is better for mobile.

Do we want to keep Google Analytics? If no, I think there is no reason to keep this plugin anymore.

The diff without the plugin can be seen here https://github.com/rspec/rspec.github.io/compare/no-plugin-from-source?expand=1
It includes a little more changes because it pull each maintenance branch before generating the doc.

## Shell commands

I added a command to remove empty lines... Ouch, there is so much... 
https://github.com/rspec/rspec.github.io/pull/139

----

All the PR are draft because I am waiting for your point of view. Especially for Google Analytics. :)

Reply:
If you want we can continue the discussion here. https://github.com/rspec/rspec-dev/pull/245